mslib/msui/topview.py

```diff
@@ -379,12 +379,16 @@ def openTool(self, index):
             self.createDockWidget(index, title, widget)
 
     def closed(self):
-        self.mainwindow_signal_login_mscolab.disconnect()
-        self.mainwindow_signal_logout_mscolab.disconnect()
-        self.mainwindow_signal_listFlighttrack_doubleClicked.disconnect()
-        self.mainwindow_signal_activate_operation.disconnect()
-        self.mainwindow_signal_permission_revoked.disconnect()
-        self.mainwindow_signal_render_new_permission.disconnect()
+        # A disconnect can be performed only once.
+        try:
+            self.mainwindow_signal_login_mscolab.disconnect()
+            self.mainwindow_signal_logout_mscolab.disconnect()
+            self.mainwindow_signal_listFlighttrack_doubleClicked.disconnect()
+            self.mainwindow_signal_activate_operation.disconnect()
+            self.mainwindow_signal_permission_revoked.disconnect()
+            self.mainwindow_signal_render_new_permission.disconnect()
+        except TypeError as ex:
+            logging.debug("mscolab connection already closed: %s", ex)
```

Comment:
These `.disconnect()` calls disconnect *all* slots that are connected to the respective signals, which belong to a different object. I.e. when the topview window is closed all slots, even from different windows, that are connected to the mainwindow's signals get disconnected. This seems wrong to me. What is the intention of this code?

Reply:
when you use the multiple flightpath docking widget with mscolab, you have one connect done  to one server. There you use then different operations over this connection.

At the moment the logout in the profile is used, a disconnect to the server is driven and this can only be done once.

Reply:
the function is not meant for closing the topview.

Reply:
> With the section removed the example shows that the second change may be enough:

Do I understand that right, that the entire method can be removed and the test passes?

> the function is not meant for closing the topview.

Right, I miss-interpreted its name. It's called when the parent view closes, at which point it also makes little sense to disconnect all slots connected to signals on the mscolab mainwindow, IMO.

Reply:
_Do I understand that right, that the entire method can be removed and the test passes?_

good idea, I try later.  I meant when I remove my changes, I can create the peek gifcycle but the test fails.